### PR TITLE
Made flavors and db operations tests work in fake mode.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Added a "fake_mode" to test.conf, which gets around things like actually logging into databases if the server is in fake mode.
- Avoided sleeping during the database tests, found different ways to test users.
- Switched to name instead of id for flavors test.

### CODE REVIEW

**Comment by TimSimpsonR** [comment]
This requires https://github.com/hub-cap/reddwarf_lite/pull/62.

**Comment by hub-cap** [comment]
+1

File: tests/integration/tests/api/users.py [hunk 1]
```diff
@@ -103,7 +106,8 @@ def test_create_users_list_system(self):
     def test_delete_users(self):
         self.dbaas.users.delete(instance_info.id, self.username_urlencoded)
         self.dbaas.users.delete(instance_info.id, self.username1_urlendcoded)
-        time.sleep(5)
+        if not FAKE:
+            time.sleep(5)
```
Comment (ed-) [inline tests/integration/tests/api/users.py:110]: Well that's one way to speed things up.

**Comment by ed-** [comment]
+1

File: tests/integration/tests/api/databases.py [hunk 2]
```diff
@@ -63,7 +64,8 @@ def test_create_database(self):
         databases.append({"name": self.dbname2})
 
         self.dbaas.databases.create(instance_info.id, databases)
-        time.sleep(5)
+        if not FAKE:
```
Comment (rnirmal) [inline tests/integration/tests/api/databases.py:67]: well I guess we can remove these and add a poll.. I suppose later..

**Comment by rnirmal** [comment]
+1

**Pull request merged**